packages/tracing/test/hub.test.ts

```diff
@@ -183,19 +178,80 @@ describe('Hub', () => {
         expect(transaction.sampled).toBe(false);
       });
 
-      it('should not sample transactions when tracesSampleRate is 0', () => {
+      it('should set sampled = false if tracesSampleRate is 0', () => {
         const hub = new Hub(new BrowserClient({ tracesSampleRate: 0 }));
         const transaction = hub.startTransaction({ name: 'dogpark' });
 
         expect(transaction.sampled).toBe(false);
       });
 
-      it('should sample transactions when tracesSampleRate is 1', () => {
+      it('should set sampled = true if tracesSampleRate is 1', () => {
         const hub = new Hub(new BrowserClient({ tracesSampleRate: 1 }));
         const transaction = hub.startTransaction({ name: 'dogpark' });
 
         expect(transaction.sampled).toBe(true);
       });
+
+      it("should call tracesSampler if it's defined", () => {
+        const tracesSampler = jest.fn();
+        const hub = new Hub(new BrowserClient({ tracesSampler }));
+        hub.startTransaction({ name: 'dogpark' });
+
+        expect(tracesSampler).toHaveBeenCalled();
+      });
+
+      it('should set sampled = false if tracesSampler returns 0', () => {
+        const tracesSampler = jest.fn().mockReturnValue(0);
+        const hub = new Hub(new BrowserClient({ tracesSampler }));
+        const transaction = hub.startTransaction({ name: 'dogpark' });
+
+        expect(tracesSampler).toHaveBeenCalled();
+        expect(transaction.sampled).toBe(false);
+      });
+
+      it('should set sampled = true if tracesSampler returns 1', () => {
+        const tracesSampler = jest.fn().mockReturnValue(1);
+        const hub = new Hub(new BrowserClient({ tracesSampler }));
+        const transaction = hub.startTransaction({ name: 'dogpark' });
+
+        expect(tracesSampler).toHaveBeenCalled();
+        expect(transaction.sampled).toBe(true);
+      });
+
+      it('should not try to override positive sampling decision provided in transaction context', () => {
+        // so that the decision otherwise would be false
+        const tracesSampler = jest.fn().mockReturnValue(0);
+        const hub = new Hub(new BrowserClient({ tracesSampler }));
+        const transaction = hub.startTransaction({ name: 'dogpark', sampled: true });
+
+        expect(transaction.sampled).toBe(true);
+      });
+
+      it('should not try to override negative sampling decision provided in transaction context', () => {
+        // so that the decision otherwise would be true
+        const tracesSampler = jest.fn().mockReturnValue(1);
+        const hub = new Hub(new BrowserClient({ tracesSampler }));
+        const transaction = hub.startTransaction({ name: 'dogpark', sampled: false });
+
+        expect(transaction.sampled).toBe(false);
+      });
+
+      it('should prefer tracesSampler to tracesSampleRate', () => {
```

Comment:
[M] I'm not sure how this test proves that. Maybe set `tracesSampleRate` to `0` and assert on `tracesSampler` being called _and_ `transaction.sampled = true`?

Reply:
Sure - can do that. I think you're right that it makes a stronger case.